pkg/types/chain_reader.go

```diff
@@ -44,16 +44,17 @@ type ChainReader interface {
 	// Note that implementations should ignore extra fields in params that are not expected in the call to allow easier
 	// use across chains and contract versions.
 	// Similarly, when using a struct for returnVal, fields in the return value that are not on-chain will not be set.
-	GetLatestValue(ctx context.Context, contractName, method string, params, returnVal any) error
+	GetLatestValue(ctx context.Context, bindingKey string, params, returnVal any) error
```

Comment:
LGTM, I am just wondering if its better UX for `GetLatestValue` to just accept `boundContract BoundContract`, I assume that products will store `BoundContract` anyway, so what is the point of the caller having to do `boundContract.Key()` instead of just `boundContract`

Reply:
I was thinking that too, but `BoundContract` also has `Pending`, which doesn't really make sense in this context. Maybe that's ok, but it felt kinda weird to me.

Reply:
Do we need to worry about pending?  Eg: Could someone subscribe to events twice, once to know it's on chain with but hasn't hit finality than again when it has finality?  Not sure the use case, but if so, then the pending flag could be ok to leave here...

Reply:
What would be the expected behavior of `Pending` then in the context of `GetLatestValue`? I'm just hesitant to include inputs that don't do anything. It makes for a confusing UX.

Reply:
Didn't we decide to remove pending? IIRC it was only added due to a miscommunication and is not actually used.

Reply:
I have an open PR that removes Pending, its just blocked atm

Reply:
Ahhh. Ok so in that case we don't have to consider `Pending` because we are eliminating it anyway.